File: Practice_DSA/quickSort.py
def quick_sort(array):
    def _quick_sort(arr, low, high):
        if low < high:
            pivot_index = (low+high)/2
            pivot_value = arr[pivot_index]

            partition_index = _partition(arr,low, high, pivot_value):

            _quick_sort(arr, low, partition_index - 1)
            _quick_sort(arr, partition_index, high)

def _partition(arr, low, high, pivot_value):
    while low <= high:
        while arr[low] < pivot_value:
            low += 1
        while arr[high] > pivot_value:
            high -= 1
        if low <= high:
            arr[low], arr[high] = arr[high], arr[low]
            low += 1
            high -= 1
        return low

_quick_sort(arr, 0, len(arr)-1)

if __name__ == "__main__":
    data = [3, 6, 8, 10, 1, 2, 1]
    print("Original array:", data)
    quick_sort(data)
    print("Sorted array:", data)

File: Practice_DSA/quickSort.cpp
#include<bits/stdc++.h>
using namespace std;

void partition(vector<int> &arr[], int low, int high){
  int pivot=arr[low];
  int i = low;
  int j = high;
  while(i<j){
    while(arr[i]<=pivot && i<=high-1){
      i++;
    }
    while(arr[j]>pivot && j>=low+1){
      j--;
    }
    if(i>j){
      swap(arr[i], arr[j]);
    }
    swap[arr[low],arr[j]];
    return j;
  }
}

void qs(vector<int> &arr[], int low, int high){
  if(low<high){
    int pIndex=partition(arr,low,mid,high);
    qs(arr,low,pIndex-1);
    qs(arr,pIndex+1,high);
  }
  
}

void quickSort(vector<int> &arr[], int n){
  qs(arr,0,n-1);
}

int main() {
    int n;
    cin>>n;
    vector<int> arr(n);
    for(int i=0;i<n;i++){
      cin>>arr[i];
    }
    quickSort(arr,n);
    for(int i=0;i<n;i++){
      cout<<arr[i]<<" ";
    }
    return 0;
  }
